### DISCUSSION TITLE
Simulation error

### DISCUSSION THREAD

**Discussion by BoburaII** [LAPKB/Pmetrics#177] [Q&A]
Hi,
I’m trying to run a simulation using a PKPD model and data from a previous PM_result object. The model and data files have 2 drug inputs and 3 output equations. I’m using Pmetrics version 2.0.2. and the simulation terminates in an error. 

Code used:

run7 <- PM_load(7) 
Sim7 <- run7$sim(limits=NA, nsim = 500, predInt=1)

Error message:

At line 8459 of file SIMeng_114.f (unit = 5, file = 'stdin')
Fortran runtime error: End of file

Error termination. Backtrace:
#0  0x17530eff7
#1  0x17530fba7
#2  0x175310483
#3  0x1753f9c33

Any suggestions on a way around this? 

Thanks

**Comment by Siel** [comment]
Hi, thank you for the report! I’ve seen that error before and put a patch
for windows users on version 2.1.0.

Are you using windows? If yes, give it a try to 2.1 and let me know

***@***.***”) (you might have to write the
command yourself to make it work)

Let me know how it goes!

On Thu, Jul 27, 2023 at 8:01 PM BoburaII ***@***.***> wrote:

> Hi,
> I’m trying to run a simulation using a PKPD model and data from a previous
> PM_result object. The model and data files have 2 drug inputs and 3 output
> equations. I’m using Pmetrics version 2.0.2. and the simulation terminates
> in an error.
>
> Code used:
>
> run7 <- PM_load(7)
> Sim7 <- run7$sim(limits=NA, nsim = 500, predInt=1)
>
> Error message:
>
> At line 8459 of file SIMeng_114.f (unit = 5, file = 'stdin')
> Fortran runtime error: End of file
>
> Error termination. Backtrace:
> #0 0x17530eff7
> #1 <https://github.com/LAPKB/Pmetrics/pull/1> 0x17530fba7
> #2 <https://github.com/LAPKB/Pmetrics/issues/2> 0x175310483
> #3 <https://github.com/LAPKB/Pmetrics/pull/3> 0x1753f9c33
>
> Any suggestions on a way around this?
>
> Thanks
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/LAPKB/Pmetrics/discussions/177>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAHZYHXF4EWWILJQDMHYSVTXSMFPLANCNFSM6AAAAAA222FRO4>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>
-- 
Enviado con Gmail para iPhone

**Reply by BoburaII** [reply]
Hi @Siel, thanks for the advice. I'm using a Mac (R recently updated to v4.3.1). An upgrade to Pmetrics version 2.1.0 sorts out the error. 

Simulation now runs to completion but only outputs 1 (pk) and 3 (pd) are being simulated. output 2 (pk) not stimulated.

A plot of output 2 returns a horizontal line through 0
code: sim7$plot(at=1, log = FALSE, outeq = 2)

Any ideas?

Again, thanks.

**Comment by Siel** [comment]
I’m glad it worked!

Regarding your second question, I don’t have enough information to answer
it. It seems the simulation is working so I’d check your model definition.

On Fri, Jul 28, 2023 at 9:02 AM BoburaII ***@***.***> wrote:

> Hi @Siel <https://github.com/Siel>, thanks for the advice. I'm using a
> Mac (R recently updated to v4.3.1). An upgrade to Pmetrics version 2.1.0
> sorts out the error.
>
> Simulation now runs to completion but only outputs 1 (pk) and 3 (pd) are
> being simulated. output 2 (pk) not stimulated.
>
> A plot of output 2 returns a horizontal line through 0
> code: sim7$plot(at=1, log = FALSE, outeq = 2)
>
> Any ideas?
>
> Again, thanks.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/LAPKB/Pmetrics/discussions/177#discussioncomment-6576470>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAHZYHVK6AUU2QXDWB53S2TXSPA53ANCNFSM6AAAAAA222FRO4>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
Enviado con Gmail para iPhone
